python/cudf/cudf/core/column_accessor.py

```diff
@@ -50,6 +108,7 @@ def insert(self, name, value, loc=-1):
         new_keys.insert(loc, name)
         new_values.insert(loc, value)
```

Comment:
Definitely a premature optimization but just wanted to note it, we could likely avoid materializing a list here by building a generator for each the keys and values that tracks the count, and when the count is `loc` we can yield `name` / `value` instead of the next value in the iterators.

This also allows us to use the cached properties we have already.